From 934cf097b98ab042a70dc625349612ddee2abcd1 Mon Sep 17 00:00:00 2001
From: Hunt Xu <mhuntxu@gmail.com>
Date: Wed, 27 Sep 2017 15:25:39 +0800
Subject: [PATCH 114/117] Sync #132 from devel to testing

db: fix backref cascade between router and es_meter_label

Fixes: redmine #11042

Signed-off-by: Hunt Xu <mhuntxu@gmail.com>
(cherry picked from commit f7cceedc098150ed05a25c34f964bb4d74e93323)
Signed-off-by: Hunt Xu <mhuntxu@gmail.com>
---
 neutron/db/metering/es_metering_db.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/neutron/db/metering/es_metering_db.py b/neutron/db/metering/es_metering_db.py
index 1876ef28b..0a4fcd8d3 100644
--- a/neutron/db/metering/es_metering_db.py
+++ b/neutron/db/metering/es_metering_db.py
@@ -45,7 +45,8 @@ class EsMeteringLabel(model_base.BASEV2, models_v2.HasId, models_v2.HasTenant):
     tcp_port = sa.Column(sa.Integer)
     router = orm.relationship(
         Router,
-        backref=orm.backref("es_metering_labels", lazy='joined', uselist=True)
+        backref=orm.backref("es_metering_labels", lazy='joined', uselist=True,
+                            cascade='all, delete-orphan')
     )
 
 
-- 
2.13.5 (Apple Git-94)

